src/inventory.h

```diff
@@ -143,13 +143,12 @@ struct ItemStack
 	// If cannot be added at all, returns the item back.
 	// If can be added partly, decremented item is returned back.
 	// If can be added fully, empty item is returned.
-	ItemStack addItem(const ItemStack &newitem,
-			IItemDefManager *itemdef);
+	ItemStack addItem(ItemStack newitem, IItemDefManager *itemdef);
 
 	// Checks whether newitem could be added.
 	// If restitem is non-NULL, it receives the part of newitem that
 	// would be left over after adding.
-	bool itemFits(const ItemStack &newitem,
+	bool itemFits(ItemStack newitem,
```

Comment:
https://github.com/minetest/minetest/pull/6039/files#diff-a101f9640ea015570f54223c277a893aL310
This param is modified in the function and used as the return value. It's nonsense to make it a const reference just to copy it.